src/App.tsx

```diff
@@ -25,19 +25,22 @@ export const App = () => {
   const [todos, setTodos] = useState(todosAndUsers);
   const [title, setTitle] = useState('');
   const [todoUserId, setTodoUserId] = useState(0);
-  const [hasError, setHasError] = useState(false);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserError, setHasUserError] = useState(false);
 
   const reset = () => {
     setTitle('');
     setTodoUserId(0);
-    setHasError(false);
+    setHasTitleError(false);
+    setHasUserError(false);
   };
 
-  const onFormSubmit = (event: React.FormEvent) => {
+  const handleSubmit = (event: React.FormEvent) => {
     event.preventDefault();
 
-    if (!todoUserId || !title) {
-      setHasError(true);
+    if (!todoUserId || !title.trim()) {
+      setHasTitleError(true);
+      setHasUserError(true);
     } else {
```

Comment:
Better immediately return from function if you have errors. Using else is bad practice.

Reply:
without else i can't pass tests
![image](https://user-images.githubusercontent.com/109803080/199797756-1949236f-6460-4604-9194-294e80532375.png)

Reply:
@rstarovoit The reason of this that you need to improve your logic here. Look at the example below:
```js
if (someErrorCondition1 || someErrorCondition2) {
  setError1(someErrorCondition1);
  setError2(someErrorCondition2);
  
  return;
}

makeSomethingIfOk();
```

Reply:
Many thanks for the example and explanations!